zkevm-circuits/src/rlp_circuit.rs

```diff
@@ -0,0 +1,2368 @@
+//! Circuit implementation for RLP-encoding verification. Please refer: https://hackmd.io/@rohitnarurkar/S1zSz0KM9.
+
+use std::marker::PhantomData;
+
+use eth_types::Field;
+use gadgets::{
+    comparator::{ComparatorChip, ComparatorConfig, ComparatorInstruction},
+    less_than::{LtChip, LtConfig, LtInstruction},
+    util::sum,
+};
+use halo2_proofs::{
+    circuit::{Layouter, Region, SimpleFloorPlanner, Value},
+    plonk::{Advice, Circuit, Column, ConstraintSystem, Error, Expression, Fixed, VirtualCells},
+    poly::Rotation,
+};
+
+use crate::{
+    evm_circuit::{
+        util::{and, constraint_builder::BaseConstraintBuilder, not, or},
+        witness::{RlpTxTag, RlpWitnessGen, N_TX_TAGS},
+    },
+    table::RlpTable,
+    util::{power_of_randomness_from_instance, Expr},
+    witness::{RlpDataType, SignedTransaction},
+};
+
+#[derive(Clone, Debug)]
+/// Config for the RLP circuit.
+pub struct RlpCircuitConfig<F> {
+    /// Denotes whether or not the row is enabled.
+    q_usable: Column<Fixed>,
+    /// Denotes whether the row is the first row in the layout.
+    is_first: Column<Advice>,
+    /// Denotes whether the row is the last byte in the RLP-encoded data.
+    is_last: Column<Advice>,
+    /// Embedded lookup table in RLP circuit.
+    rlp_table: RlpTable,
+    /// Denotes the index of this row, starting from `1` and ending at `n`
+    /// where `n` is the byte length of the RLP-encoded data.
+    index: Column<Advice>,
+    /// Denotes the index of this row, but reversed. It starts from `n` where
+    /// `n` is the byte length of the RLP-encoded data and ends at `1`.
+    rindex: Column<Advice>,
+    /// Denotes the byte value at this row index from the RLP-encoded data.
+    byte_value: Column<Advice>,
+    /// Denotes the RLC accumulator value used for call data bytes.
+    value_acc_rlc: Column<Advice>,
+    /// List of columns that are assigned:
+    /// val := (tag - RlpTxTag::{Variant}).inv()
+    tx_tags: [Column<Advice>; N_TX_TAGS],
+    /// Denotes the current tag's span in bytes.
+    tag_length: Column<Advice>,
+    /// Denotes an accumulator for the length of data, in the case where len >
+    /// 55 and the length is represented in its big-endian form.
+    length_acc: Column<Advice>,
+    /// Denotes an accumulator for the value field over all rows (bytes).
+    value_rlc: Column<Advice>,
```

Comment:
How about renaming this column to `all_bytes_rlc_acc` so as to make it more clear?